File: app/src/main/java/ru/nsu/fit/modao/MasterActivity.java
package ru.nsu.fit.modao;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.navigation.NavigationBarView;

import ru.nsu.fit.modao.databinding.ActivityMasterBinding;

public class MasterActivity extends AppCompatActivity {
    ActivityMasterBinding binding;
    NavController navController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMasterBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());


        NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.fragment);
        navController = navHostFragment.getNavController();
        binding.bottomMenu.setOnItemSelectedListener(new NavigationBarView.OnItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                navController.navigate(item.getItemId());
                return true;
            }
        });
    }

}

File: app/src/main/java/ru/nsu/fit/modao/MasterActivity.kt
package ru.nsu.fit.modao

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import dagger.hilt.android.AndroidEntryPoint
import ru.nsu.fit.modao.databinding.ActivityMasterBinding
@AndroidEntryPoint
class MasterActivity: AppCompatActivity() {
    private lateinit var binding: ActivityMasterBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMasterBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.bottomMenu.setOnItemSelectedListener {item ->
            when (item.itemId) {
                R.id.nested_groups -> findNavController(R.id.fragment)
                    .navigate(R.id.action_global_nested_groups)
                R.id.notification_fragment -> findNavController(R.id.fragment)
                    .navigate(R.id.action_global_notification_fragment)
                R.id.friends_fragment -> findNavController(R.id.fragment)
                    .navigate(R.id.action_global_friends_fragment)
                R.id.profile_fragment -> findNavController(R.id.fragment)
                    .navigate(R.id.action_global_profile_fragment)
                else -> findNavController(R.id.fragment)
                    .navigate(item.itemId)
            }
            return@setOnItemSelectedListener true
        }
    }

}